cmd/generate/cmd_util.go

```diff
@@ -0,0 +1,85 @@
+package main
+
+import (
+	"fmt"
+	"strings"
+
+	"github.com/nginxinc/nginx-go-crossplane/internal/generator"
+)
+
+type filterFlag struct {
+	filter map[string]struct{}
+}
+
+func (reader *filterFlag) Set(value string) error {
+	if reader.filter == nil {
+		reader.filter = make(map[string]struct{})
+	}
+	reader.filter[value] = struct{}{}
+	return nil
+}
+
+func (reader *filterFlag) String() string {
+	return fmt.Sprintf("%v", *reader)
+}
+
+type overrideParser struct {
+	directive string
+	masks     []generator.Mask
+}
+
+//nolint:mnd
+func (override *overrideParser) UnmarshalText(text []byte) error {
+	rawOverride := string(text)
+
+	// rawStr should follow the format: directive:bitmask00|bitmask01|...,bitmask10|bitmask11|...
+	subs := strings.Split(rawOverride, ":")
+	if len(subs) != 2 {
+		return fmt.Errorf("override parameter %s format wrong", rawOverride)
+	}
+
+	directive := strings.TrimSpace(subs[0])
+	override.directive = directive
+	if directive == "" {
+		return fmt.Errorf("directive name is empty:override=%s", rawOverride)
+	}
+
+	if strings.TrimSpace(subs[1]) == "" {
+		return fmt.Errorf("directive masks is empty:override=%s", rawOverride)
+	}
+
+	for _, varNamesStr := range strings.Split(subs[1], ",") {
+		varNamesList := strings.Split(varNamesStr, "|")
+		varNamesNum := len(varNamesList)
+		directiveMask := make(generator.Mask, varNamesNum)
+
+		for idx, varName := range varNamesList {
+			trimmedName := strings.TrimSpace(varName)
+			if trimmedName == "" {
+				return fmt.Errorf("one directive bitmasks is empty, check if there are unnecessary |:override=%s", rawOverride)
+			}
+
+			directiveMask[idx] = trimmedName
+		}
+		override.masks = append(override.masks, directiveMask)
+	}
+
+	return nil
+}
+
+type overrideParserList []overrideParser
+
+func (overrides *overrideParserList) String() string {
+	return fmt.Sprintf("%v", *overrides)
+}
+
+func (overrides *overrideParserList) Set(value string) error {
+	var parser overrideParser
+	err := parser.UnmarshalText([]byte(value))
+	if err != nil {
+		return err
```

Comment:
nit: we can include the raw value here and simplify some of the error building in `UnmarshalText`

```suggestion
		return fmt.Errorf("invalid override %q: %w", value, err)
```

Reply:
Fixed: changed error return format